Sprint-2/errors/0.js

```diff
@@ -4,6 +4,64 @@
 // interpret the error message and figure out why an error is occurring
 
 function capitalise(str) {
-  let str = `${str[0].toUpperCase()}${str.slice(1)}`;
-  return str;
+  let strOut = `${str[0].toUpperCase()}${str.slice(1)}`;
+  return strOut;
 }
+
+console.log(`Test with normal data: "hello guys" is ${capitalise("hello guys")}`);
+console.log(`Test with abnormal data: "h" is ${capitalise("h")}`);
+//console.log(`Test with abnormal data: "" is ${capitalise("")}`);
+//console.log(`Test with extreme data: 234 is ${capitalise(234)}`);
+console.log(`Test with normal data: "hello guys again" is ${capitalise("hello guys again")}`);
+
+/* 
+My prediction:
+I will give the prediction in the order of theories, 
+i.e. what I noticed first, etc. I will only analyze line 7:
+let str = `${str[0].toUpperCase()}${str.slice(1)}`;
+because there is nothing else to analyze
+
+1. Judging by the code, we should accept the incoming string 
+and return the same string, but converting the first character 
+of the string to uppercase.
+
+2. The first potential error is if the incoming string is empty 
+or one character long, i.e. the minimum length of the incoming string 
+should be at least 2, otherwise it will cause a critical error. 
+Maybe not critical, I don't know node.JS behavior that well yet
+
+3. We should have an error because we redefine the "let str" 
+variable with the same identifier as the parameter of our function. 
+If there is no error in this place, then the JS interpreter 
```

Comment:
Hee! Love it

Reply:
I won't hide it, I sometimes make such mistakes myself, so I found it here easily and quickly 😉